src/GafferScene/OptionTweaks.cpp

```diff
@@ -0,0 +1,148 @@
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "GafferScene/OptionTweaks.h"
+
+#include "Gaffer/TweakPlug.h"
+
+using namespace std;
+using namespace IECore;
+using namespace Gaffer;
+using namespace GafferScene;
+
+GAFFER_NODE_DEFINE_TYPE( OptionTweaks );
+
+size_t OptionTweaks::g_firstPlugIndex = 0;
+
+OptionTweaks::OptionTweaks( const std::string &name ) : GlobalsProcessor( name )
+{
+	storeIndexOfNextChild( g_firstPlugIndex );
+	addChild( new BoolPlug( "ignoreMissing", Plug::In, false ) );
+	addChild( new TweaksPlug( "tweaks" ) );
+}
+
+OptionTweaks::~OptionTweaks()
+{
+
+}
+
+Gaffer::BoolPlug *OptionTweaks::ignoreMissingPlug()
+{
+	return getChild<Gaffer::BoolPlug>( g_firstPlugIndex );
+}
+
+const Gaffer::BoolPlug *OptionTweaks::ignoreMissingPlug() const
+{
+	return getChild<Gaffer::BoolPlug>( g_firstPlugIndex );
+}
+
+Gaffer::TweaksPlug *OptionTweaks::tweaksPlug()
+{
+	return getChild<Gaffer::TweaksPlug>( g_firstPlugIndex + 1 );
+}
+
+const Gaffer::TweaksPlug *OptionTweaks::tweaksPlug() const
+{
+	return getChild<Gaffer::TweaksPlug>( g_firstPlugIndex + 1 );
+}
+
+void OptionTweaks::affects( const Plug *input, AffectedPlugsContainer &outputs ) const
+{
+	GlobalsProcessor::affects( input, outputs );
+
+	if( tweaksPlug()->isAncestorOf( input ) || input == ignoreMissingPlug() )
+	{
+		outputs.push_back( outPlug()->globalsPlug() );
+	}
+}
+
+void OptionTweaks::hashProcessedGlobals( const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	if( tweaksPlug()->children().empty() )
+	{
+		h = inPlug()->globalsPlug()->hash();
+	}
+	else
+	{
+		ignoreMissingPlug()->hash( h );
+		tweaksPlug()->hash( h );
+	}
+}
+
+IECore::ConstCompoundObjectPtr OptionTweaks::computeProcessedGlobals(
+	const Gaffer::Context *context,
+	const IECore::ConstCompoundObjectPtr inputGlobals
+) const
+{
+	const TweaksPlug *tweaksPlug = this->tweaksPlug();
+	if( tweaksPlug->children().empty() )
+	{
+		return inputGlobals;
+	}
+
+	const bool ignoreMissing = ignoreMissingPlug()->getValue();
+
+	CompoundObjectPtr result = new CompoundObject();
+	result->members() = inputGlobals->members();
+
+	const CompoundObject *source = inputGlobals.get();
+
+	const std::string prefix = namePrefix();
+
+	tweaksPlug->applyTweaks(
+		[&source, &prefix]( const std::string &valueName )
+		{
+			return source->member<Data>( prefix + valueName );
+		},
+		[&result, &prefix]( const std::string &valueName, DataPtr newData )
+		{
+			if( newData == nullptr )
+			{
+				return result->members().erase( prefix + valueName ) > 0;
+			}
+			result->members()[prefix + valueName] = newData;
+			return true;
+		},
+		ignoreMissing ? TweakPlug::MissingMode::Ignore : TweakPlug::MissingMode::Error
+	);
+
+	return result;
+}
+
+const std::string OptionTweaks::namePrefix() const
+{
+	return "option:";
+}
```

Comment:
I had also thought it may be useful to extend to tweaking / querying global attributes, but it sounds like that's not a big concern, and it's easy to add later. Removed in f85906f8c6c981ab908f78855b1f1e1e029a2c09

Out of curiosity, what are the global attributes usually used for?

Reply:
> I had also thought it may be useful to extend to tweaking / querying global attributes

That's an interesting one - perhaps user expectation would be that we'd add a `global` checkbox to the existing AttributeTweaks, rather than create a separate GlobalAttributeTweaks, so that we're mirroring the behaviour of StandardAttributes?

> Out of curiosity, what are the global attributes usually used for?

One thing they're useful for is setting a global value for the number of motion samples. That can be varied per-object so it's an attribute rather than an option, but it is also pretty useful to be able to set it globally.